cranelift/src/souper_harvest.rs

```diff
@@ -61,24 +58,8 @@ pub fn run(options: &Options) -> Result<()> {
         .read_to_end(&mut contents)
         .context("failed to read input file")?;
 
-    let funcs = if &contents[..WASM_MAGIC.len()] == WASM_MAGIC {
-        let mut dummy_environ = DummyEnvironment::new(
-            fisa.isa.unwrap().frontend_config(),
-            ReturnMode::NormalReturns,
-            false,
-        );
-        cranelift_wasm::translate_module(&contents, &mut dummy_environ)
-            .context("failed to translate Wasm module to clif")?;
-        dummy_environ
-            .info
-            .function_bodies
-            .iter()
-            .map(|(_, f)| f.clone())
-            .collect()
-    } else {
-        let contents = String::from_utf8(contents)?;
-        cranelift_reader::parse_functions(&contents)?
-    };
+    let contents = String::from_utf8(contents)?;
+    let funcs = cranelift_reader::parse_functions(&contents)?;
```

Comment:
This will make it much harder to use souper_harvest for wasm modules.

Reply:
Yeah, ideally we would still be able to harvest LHSes from Wasm modules here. Or we would move the Wasm harvester into the `wasmtime` CLI and leave this as a CLIF-only harvestor.

Reply:
FWIW @fitzgen I'll leave this in since I'll back out the changes to `wasmtime-cranelift`, but I think if we want to harvest things from wasm that Wasmtime uses then this is the wrong level of abstraction becaue it's missing all of Wasmtime's configuration of Cranelift as well as Wasmtime-specific implementation details of runtime things like tables/references/etc. 

I'll admit I don't know why this command exists, I've never actually seen anyone use it. It also seems to rely somewhat heavily on the `Dummy*` traits in `cranelift-wasm` which are barely maintained at best and shouldn't be a great source of "we're getting real data from these"

Reply:
Yeah, is has known issues around using a different environment than Wasmtime. That said, most of the interesting stuff from a superoptimizer's point of view are the expressions, which don't typically change much depending on the environment, not the I/O and state integration code that do change a bunch depending on the environment.